django/contrib/contenttypes/fields.py

```diff
@@ -243,7 +241,8 @@ def __get__(self, instance, cls=None):
                 rel_obj = ct.get_object_for_this_type(pk=pk_val)
             except ObjectDoesNotExist:
                 pass
-        setattr(instance, self.cache_attr, rel_obj)
+
+        self.set_cached_value(instance=instance, value=rel_obj)
```

Comment:
I thought about it, the reason I do it like this is because there are many cases where instance is the value and something else like rel_obj is the instance where the cache is being set. This can look very confusing and inconsistent (in my opinion) if we use positional args.